user/src/com/google/web/bindery/autobean/gwt/rebind/AutoBeanFactoryGenerator.java

```diff
@@ -372,12 +372,25 @@ private void writeEnumSetup(SourceWriter sw) {
       list.add(entry.getKey());
     }
 
-    sw.println("@Override protected void initializeEnumMap() {");
+    int methodCount = 0;
+    int enumPerMethodCount = 0;
+    int maxEnumsPerMethod = 2000;
+    sw.println("private void initializeEnumMap_%d() {", methodCount);
     sw.indent();
     for (Map.Entry<JEnumConstant, String> entry : model.getEnumTokenMap().entrySet()) {
       // enumToStringMap.put(Enum.FOO, "FOO");
       sw.println("enumToStringMap.put(%s.%s, \"%s\");", entry.getKey().getEnclosingType()
           .getQualifiedSourceName(), entry.getKey().getName(), entry.getValue());
+
+      enumPerMethodCount++;
+      if (enumPerMethodCount >= maxEnumsPerMethod) {
+        sw.outdent();
+        sw.println("}");
+        enumPerMethodCount = 0;
+        methodCount++;
+        sw.println("private void initializeEnumMap_%d() {", methodCount);
+        sw.indent();
+      }
```

Comment:
The duplication in these two sections bothers me... but I think not enough to ask for it to be factored out into a lambda or class that keeps the code more readable.

(Unless you were looking for a nudge to clean that up further.)

Pre-submit edit: Okay two more things that bother me here (but again not enough to require a change before merge):
 * the first initializedEnumMap_N call always has a zero suffix, but still uses `%d`, which is a little silly.
 * It is possible to emit a new method at the end of a loop, thus leaving it empty. 

It could make slightly more sense to remove the first method creation call, and move each "do i need to start a new method" check to the top of the loops they belong to. Then, initialize `enumPerMethodCount` to greater than `maxEnumsPerMethod` and you'll always get an method created if needed. Last, the final outdent+println(}) can be conditional - don't print if methodCount==0.

Thoughts?

Reply:
You are correct it could have created a new method at the end of a loop, thus leaving it empty. I moved the closing of the previous method and creation of a new method to the top of the loops. 

One quick note the initializedEnumMap_N now start at 1 so the last loop to call the `initializeEnumMap()` now starts at 1 too.